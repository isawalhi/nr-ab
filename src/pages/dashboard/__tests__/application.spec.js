import Application from '../components/application/applicationModel'
import ApplicationView from '../components/application/applicationView'

describe('Application', () => {
    const data = {
        name: 'test-app',
        contributors: ['test-cont-1'],
        version: 1,
        apdex: 89,
    }
    const application = new Application(data);

    it('assigns constructor values correctly', () => {
        expect(application.name).toBe(data.name)
        expect(application.contributors).toBe(data.contributors)
        expect(application.version).toBe(data.version)
        expect(application.apdex).toBe(data.apdex)
    })

    it('generates html when application is provided correctly', () => {
        expect(new ApplicationView().generateMarkup(application)).toMatchSnapshot()
    })

    it('return empty html when application is not provided correctly', () => {
        expect(new ApplicationView().generateMarkup(null)).toMatchSnapshot()
    })
})