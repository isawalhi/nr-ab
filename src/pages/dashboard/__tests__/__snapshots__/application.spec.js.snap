// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application generates html when application is provided correctly 1`] = `
"
        <div key=\\"test-app\\" release-version=\\"1\\" class=\\"application\\">
            <div class=\\"app-apdex\\">89</div>
            <div class=\\"app-name\\">test-app</div>
        </div>
        "
`;

exports[`Application return empty html when application is not provided correctly 1`] = `""`;
